### PR TITLE
[MU4] fix #8694 grip dragging not working

### LINKED ISSUE #8694: Lines not resizing correctly
Line items from the palette can't be lengthened or shortened in the score/project editor.

This seems to be affecting ALL line objects (e.g. slurs, hairpins, crescendo lines, pedal lines, etc.... albeit with **some** inconsistency, as you will see in this demo video):
 
https://user-images.githubusercontent.com/86290556/126642964-de6b6bb1-79ae-4bd0-8aee-fabe50db18bf.mp4

**Expected behaviour**
User should be able to lengthen/shorten all line objects by adjusting left and right handles. 
Line objects should also snap to notes.
See for example: 
https://user-images.githubusercontent.com/86290556/126642985-aa6e2405-2620-4c8c-b741-184b400a2197.mov

Apple MacBook Air M1 MacOS 11.4
Reproduced in Windows 10 ARM (via Parallels)

### PR DESCRIPTION
**Pull request by wizofaus**
Resolves: https://github.com/musescore/issues/8694

Grip mouse operations not working

- [x] I signed [CLA](https://musescore.org/en/cla)
- [x] I made sure the code in the PR follows [the coding rules](https://github.com/musescore/MuseScore/wiki/CodeGuidelines)
- [x] I made sure the code compiles on my machine
- [x] I made sure there are no unnecessary changes in the code
- [x] I made sure the title of the PR reflects the core meaning of the issue you are solving
- [x] I made sure the commit message(s) contain a description and answer the question "Why do those changes fix that particular issue?" or "Why are those changes really necessary as improvements?"
- [x] I made sure the commit message title starts with "fix #424242:" if there is a related issue
- [ ] I created the test (mtest, vtest, script test) to verify the changes I made

### CODE REVIEW

File: src/notation/internal/notationinteraction.cpp [hunk 1]
```diff
@@ -2976,15 +2972,10 @@ void NotationInteraction::updateGripEdit(const std::vector<Element*>& elements)
...
         return;
     }
-
     m_gripEditData.grips = element->gripsCount();
     m_gripEditData.curGrip = Ms::Grip::NO_GRIP;
     m_gripEditData.element = element;
-    m_gripEditData.grip.resize(m_gripEditData.grips);
-
     m_gripEditData.element->startEdit(m_gripEditData);
-    m_gripEditData.element->updateGrips(m_gripEditData);
```
Comment (wizofaus) [inline src/notation/internal/notationinteraction.cpp:2986]: this was producing meaningless grip rectangles that weren't adjusted for current world transform at all

File: src/notation/internal/notationinteraction.cpp [hunk 2]
```diff
@@ -2094,56 +2094,52 @@ bool NotationInteraction::isGripEditStarted() const
...
+            return i;
         }
     }
+    return -1;
+}
 
-    return false;
+bool NotationInteraction::isHitGrip(const PointF& pos) const
+{
+    return selection()->element() && findGrip(m_gripEditData.grip, point2page(pos), pos) != -1;
```
Comment (RomanPudashkin) [inline src/notation/internal/notationinteraction.cpp:2114]: You don't use selection()->element() here, so you can remove it from here

Comment (wizofaus) [reply src/notation/internal/notationinteraction.cpp:2114]: `isHitGrip` should return `false` if  `selection()->element()` is `nullptr`.
In principle `m_gripEditData.grip` should always be an empty array in that case, but the check was there already.

**Review by RomanPudashkin — APPROVED** [review]

**Comment by RobFog** [comment]
Link above gives a 404. This should be the issue that was fixed: #8694.

**Comment by vshalkevich** [comment]
Works fine

**Pull request merged**